src/main/scala/ch/epfl/bluebrain/nexus/admin/organizations/OrganizationState.scala

```diff
@@ -0,0 +1,113 @@
+package ch.epfl.bluebrain.nexus.admin.organizations
+import java.time.Instant
+import java.util.UUID
+
+import cats.effect.Async
+import ch.epfl.bluebrain.nexus.admin.config.AppConfig.HttpConfig
+import ch.epfl.bluebrain.nexus.admin.config.Vocabulary.nxv
+import ch.epfl.bluebrain.nexus.admin.organizations.OrganizationCommand._
+import ch.epfl.bluebrain.nexus.admin.organizations.OrganizationEvent._
+import ch.epfl.bluebrain.nexus.admin.types.{ResourceF, ResourceMetadata}
+import ch.epfl.bluebrain.nexus.commons.types.identity.Identity
+
+/**
+  * Enumeration of organization states.
+  */
+sealed trait OrganizationState extends Product with Serializable
+
+object OrganizationState {
+
+  /**
+    * The initial (undefined) state.
+    */
+  final case object Initial extends OrganizationState
+
+  /**
+    * Initial organization state.
+    *
+    * @param organization the organization representation
+    * @param uuid         the permanent identifier of the organization
+    * @param rev          the organization revision
+    * @param deprecated   the deprecation status of the organization
+    * @param createdAt    the instant when the organization was created
+    * @param updatedAt    the instant when the organization was last updated
+    * @param createdBy    the identity that created the organization
+    * @param updatedBy    the identity that last updated the organization
+    */
+  final case class Current(organization: Organization,
+                           uuid: UUID,
+                           rev: Long,
+                           deprecated: Boolean,
+                           createdAt: Instant,
+                           updatedAt: Instant,
+                           createdBy: Identity,
+                           updatedBy: Identity)
+      extends OrganizationState {
+
+    def toResource(implicit http: HttpConfig): ResourceF[Organization] =
+      ResourceF(http.baseIri + "orgs" + organization.label,
+                uuid,
+                rev,
+                deprecated,
+                Set(nxv.Organization),
+                createdAt,
+                createdBy,
+                updatedAt,
+                updatedBy,
+                organization)
+
+    def toResourceMetadata(implicit http: HttpConfig): ResourceMetadata =
+      ResourceF.unit(http.baseIri + "orgs" + organization.label,
+                     uuid,
+                     rev,
+                     deprecated,
+                     Set(nxv.Organization),
+                     createdAt,
+                     createdBy,
+                     updatedAt,
+                     updatedBy,
+      )
+  }
+
+  def next(state: OrganizationState, ev: OrganizationEvent): OrganizationState = (state, ev) match {
```

Comment:
I would define the `next` and `evaluate` in the `Organizations[F]` companion object and make them private.